From 1420e204b0bcb8f30fb054828cd71d17b2ffe379 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Fri, 27 Jan 2023 17:31:38 +0000
Subject: [PATCH] Fix build with nodejs-18

:Release Notes:
Fixes:
http://gecko.lge.com/Errors/Details/527976

:Detailed Notes:
../src/pmloglib.cpp:194:111: error: no matching function for call to 'v8::ScriptOrigin::ScriptOrigin(v8::Local<v8::String>)'
  194 |     ScriptOrigin *scriptOrigin = new ScriptOrigin(String::NewFromUtf8(isolate, "pmloglib.js").ToLocalChecked());
      |                                                                                                               ^
In file included from nodejs-module-webos-pmlog/3.0.1-7-r14/node-v18.12.1/deps/v8/include/v8-script.h:17,
                 from nodejs-module-webos-pmlog/3.0.1-7-r14/node-v18.12.1/deps/v8/include/v8-debug.h:10,
                 from nodejs-module-webos-pmlog/3.0.1-7-r14/node-v18.12.1/deps/v8/include/v8.h:29:
nodejs-module-webos-pmlog/3.0.1-7-r14/node-v18.12.1/deps/v8/include/v8-message.h:64:13: note: candidate: 'v8::ScriptOrigin::ScriptOrigin(v8::Isolate*, v8::Local<v8::Value>, int, int, bool, int, v8::Local<v8::Value>, bool, bool, bool, v8::Local<v8::Data>)'
   64 |   V8_INLINE ScriptOrigin(Isolate* isolate, Local<Value> resource_name,
      |             ^~~~~~~~~~~~
nodejs-module-webos-pmlog/3.0.1-7-r14/node-v18.12.1/deps/v8/include/v8-message.h:64:13: note:   candidate expects 11 arguments, 1 provided
nodejs-module-webos-pmlog/3.0.1-7-r14/node-v18.12.1/deps/v8/include/v8-message.h:62:17: note: candidate: 'constexpr v8::ScriptOrigin::ScriptOrigin(const v8::ScriptOrigin&)'
   62 | class V8_EXPORT ScriptOrigin {
      |                 ^~~~~~~~~~~~
nodejs-module-webos-pmlog/3.0.1-7-r14/node-v18.12.1/deps/v8/include/v8-message.h:62:17: note:   no known conversion for argument 1 from 'v8::Local<v8::String>' to 'const v8::ScriptOrigin&'
nodejs-module-webos-pmlog/3.0.1-7-r14/node-v18.12.1/deps/v8/include/v8-message.h:62:17: note: candidate: 'constexpr v8::ScriptOrigin::ScriptOrigin(v8::ScriptOrigin&&)'
nodejs-module-webos-pmlog/3.0.1-7-r14/node-v18.12.1/deps/v8/include/v8-message.h:62:17: note:   no known conversion for argument 1 from 'v8::Local<v8::String>' to 'v8::ScriptOrigin&&'

Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/nodejs-module-webos-pmlog/+/344285 Fix build with nodejs-18]
Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/pmloglib.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/pmloglib.cpp b/src/pmloglib.cpp
index 4c5dc9d..141eba6 100644
--- a/src/pmloglib.cpp
+++ b/src/pmloglib.cpp
@@ -186,7 +186,8 @@ init(Local<Object> target)
                                                        (const char*)pmloglib_js,
                                                        v8::NewStringType::kNormal,
                                                        pmloglib_js_len).ToLocalChecked();
-	ScriptOrigin *scriptOrigin = new ScriptOrigin(String::NewFromUtf8(isolate, "pmloglib.js").ToLocalChecked());
+    ScriptOrigin *scriptOrigin = new ScriptOrigin(isolate, String::NewFromUtf8(isolate, "pmloglib.js").ToLocalChecked());
+
     Local<Script> script = Script::Compile(isolate->GetCurrentContext(),
         scriptText, scriptOrigin).ToLocalChecked();
     if (!script.IsEmpty()) {
